src/components/NavigationTree/NavigationTreeNode.tsx

```diff
@@ -36,15 +39,54 @@ function renderIcon(type: NavigationTreeNodeType | string, collapsed: boolean) {
 
 export function NavigationTreeNode({
     path,
+    fetchPath,
+    activePath,
     state,
+    level,
     dispatch,
     children,
-    active,
     onActivate,
     getActions,
+    cache,
 }: NavigationTreeNodeProps) {
     const nodeState = state[path];
 
+    React.useEffect(() => {
+        if (nodeState.collapsed) {
+            if (!cache) {
+                dispatch({
+                    type: NavigationTreeActionType.ResetNode,
+                    payload: {path},
+                });
+            }
```

Comment:
the former approach with checking `loaded || !cache` on mount doesn't work with virtualization, since components outside the viewport unmount, and scrolling back to them causes open nodes to reload, which is incorrect